rocky/account/mixins.py

```diff
@@ -203,3 +206,50 @@ class OrganizationPermissionRequiredMixin(PermissionRequiredMixin):
     def has_permission(self) -> bool:
         perms = self.get_permission_required()
         return self.organization_member.has_perms(perms)
+
+
+class OrganizationAPIMixin:
+    request: Request
+
+    @cached_property
+    def organization(self) -> Organization:
+        try:
+            organization_id = self.request.query_params["organization_id"]
+        except KeyError:
+            pass
+        else:
+            try:
+                return Organization.objects.get(id=organization_id)
```

Comment:
I'd rather have this factory as a method on the user object, where the user instance limits the organizations visible/loadbable to it by checking its permissions and memberships.